packages/matter.js/src/protocol/interaction/AttributeDataDecoder.ts

```diff
@@ -162,3 +162,17 @@ export function decodeChunkedArray<T>(schema: ArraySchema<T>, values: TypeFromSc
     schema.validate(result);
     return result;
 }
+
+export function structureReadDataToClusterObject(data: DecodedAttributeReportValue[]) {
+    const structure: { [key: number]: { [key: number]: { [key: string]: any } } } = {};
+    for (const { path: { endpointId, clusterId, attributeName }, value } of data) {
+        if (structure[endpointId] === undefined) {
```

Comment:
```javascript
Object.assign(structure, {[endpointId]: {[clusterId]: {[attributeName]: value}}})
```
saves from writing too many if statements ;)

Reply:
good catch .. thx, done

Reply:
Ahhh nooooo ... this do not work!! Object assign is not working recursive with objects ...

Reply:
Yes but a depenency to add for that some lines code is it not worth ;-)